merkle/sparse_merkle_tree_test.go

```diff
@@ -66,19 +66,21 @@ func getSparseMerkleTreeReaderWithMockTX(ctrl *gomock.Controller, rev int64) (*S
 	return NewSparseMerkleTreeReader(rev, maphasher.Default, tx), tx
 }
 
-func runOnProducer(tx storage.MapTreeTX) func(context.Context, func(context.Context, storage.MapTreeTX) error) error {
-	defer tx.Close()
+type producerTXRunner struct {
+	tx storage.MapTreeTX
+}
 
-	return func(ctx context.Context, f func(context.Context, storage.MapTreeTX) error) error {
-		return f(ctx, tx)
-	}
+func (r *producerTXRunner) RunTX(ctx context.Context, f func(context.Context, storage.MapTreeTX) error) error {
+	r.tx.Close()
```

Comment:
This closes the transaction before running `f` on it?

Reply:
should be a defer.